google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -186,7 +190,12 @@ public static GapicSpannerRpc create(SpannerOptions options) {
 
   public GapicSpannerRpc(final SpannerOptions options) {
     this.projectId = options.getProjectId();
-    this.projectName = PROJECT_NAME_TEMPLATE.instantiate("project", this.projectId);
+    String projectNameStr = PROJECT_NAME_TEMPLATE.instantiate("project", this.projectId);
+    try {
+      projectNameStr = URLDecoder.decode(projectNameStr, StandardCharsets.UTF_8.toString());
```

Comment:
I know you have it in the commit, but if you can add a brief comment here on why we're doing this, that would be great.

Reply:
Done